integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/alterTable/TestAlterTableAddColumns.scala

```diff
@@ -214,6 +214,24 @@ class TestAlterTableAddColumns extends QueryTest with BeforeAndAfterAll {
     sql("DROP TABLE IF EXISTS alter_com")
   }
 
+  test("Test alter add complex type and compaction") {
```

Comment:
you can add compaction scenario in test cases which contains SI on complex and primitive columns. You can edit existing test cases only like
`test array<string> and string as index columns on secondary index` , `test multiple SI with array and primitive type`

Reply:
Done